src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -535,9 +539,134 @@ def load_sub_model(
         # else load from the root directory
         loaded_sub_model = load_method(cached_folder, **loading_kwargs)
 
+    if isinstance(loaded_sub_model, torch.nn.Module) and device_map_strategy == "balanced":
+        if needs_offloading:
+            dispatch_model(loaded_sub_model, device_map=device_map, force_hooks=True, main_device=0)
+        else:
+            dispatch_model(loaded_sub_model, device_map=device_map, force_hooks=True)
+
```

Comment:
why do we dispatch the module that are not offloded only when `device_map_strategy == "balanced" `?

Reply:
From this suggestion: https://github.com/huggingface/diffusers/pull/6857#issuecomment-1969375705. 

Is there any other way to achieve this?

Reply:
It's because for the other two strategies ("balanced_low_memory" and "balanced_ultra_low_memory"), we rely on `enable_model_cpu_offload()` and `enable_sequential_cpu_offload()`.

Reply:
Oh, so the two other strategy are not compatible with multi-gpu. Is that right ?

Reply:
From our conversation: this case is only useful when things on the same device and is not particularly useful for multi-gpu scenario.